### PR TITLE
Cryo improvements: Auto eject dead and turn on automatically

### PR DESCRIPTION
**Pull request by Momo8289**
## About The Pull Request
This PR makes two changes to cryo. First, if the patient inside is dead, it will notify medical over comms, and if auto mode is on, it will eject them. This is the exact same behaviour as if the patient is fully healed. Second, if auto is on, the tube will turn on as soon as it's closed, preventing negligent doctors from leaving people to die in a locked tube. To prevent people from putting themselves in cryo, you can no longer close the cryo tube yourself when you're standing in it. This is effectively the exact same as it was before, except now the game explicitly tells you that you cannot cryo yourself, instead of letting you jump in and die.

## Why It's Good For The Game
For the first change, there is absolutely zero reasons for a dead body to be in a cryotube, and it makes sense that a machine that can tell if someone is fully healed and eject them can tell if they're dead and eject them.

For the second change, there are few things worse than being left to die in a cryotube just because the doctor didn't turn it on. You're just locked in there until another doctor walks by and notices that the tube isn't actually on (if they ever do), and you likely die before then, leaving your organs decay, and turn what was once some simple brute and burn into a whole revival ordeal. "Auto" implies that treatment is automatic, so it makes sense that it actually is.
## Changelog
:cl:
qol: Cryotubes will now notify medbay if the patient within is dead, and will eject them if auto is on.
qol: Cryotubes will now automatically turn on when a patient enters it if auto is on, but you can no longer close the cryotube on yourself.
/:cl:

### CODE REVIEW

**Comment by Pickle-Coding** [comment]
Doctors having to turn on the cryo tube is a balance decision. If it automatically turns on, then people would be able to cryo themselves without the assistance of another player.

**Comment by Momo8289** [comment]
> Doctors having to turn on the cryo tube is a balance decision. If it automatically turns on, then people would be able to cryo themselves without the assistance of another player.

ah that's true

**Comment by Elli-Skala** [comment]
I think just the dead body eject (maybe with a medical radio message?) Alone will do just fine

It's more soulful with doctors some times just forgetting you but without the gay baby jail that comes from being dead inside them

**Comment by nukularpower** [comment]
just something to consider on this note, maybe for another PR or whatever:  it happens frequently that you are as healed as possible from cryo, but the damage from the cold keeps ticking you down below the release threshold, so you are stuck way longer than you really need to be.   Would it be possible to move the eject health value down a bit to prevent this, as it almost always happens?

**Comment by Momo8289** [comment]
> I think just the dead body eject (maybe with a medical radio message?) Alone will do just fine

There is a medical radio message, yeah.

> It's more soulful with doctors some times just forgetting you but without the gay baby jail that comes from being dead inside them 

The way I see it, its a tradeoff. Cryo looses one button press of interactivity for the medical staff, and some of that soulfulness you mentioned. In return, people don't potentially get their rounds ruined by a doctor who forgot to press a button. In my opinion it's worth it. 

I'm fine with not including it if people thing the death eject is enough, but I really think it should be considered.

**Comment by Momo8289** [comment]
> just something to consider on this note, maybe for another PR or whatever:  it happens frequently that you are as healed as possible from cryo, but the damage from the cold keeps ticking you down below the release threshold, so you are stuck way longer than you really need to be.   Would it be possible to move the eject health value down a bit to prevent this, as it almost always happens?

My friend mentioned a similar thing with cryo screwing you over if you have prosthetics, so I might make another PR about that later.

**Comment by carlarctg** [comment]
good change, some ridiculous abstract notion of PLAYER ENGAGEMENT and SOULE isn't more important than not letting players be trapped in hell forever

you could just force eject anyone after x minutes and make that a slider on the cryobtube UI, this would fix all issues without upsetting soul losers

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 1]
```diff
@@ -383,6 +402,9 @@
 /obj/machinery/atmospherics/components/unary/cryo_cell/close_machine(mob/living/carbon/user, density_to_set = TRUE)
 	treating_wounds = FALSE
 	if((isnull(user) || istype(user)) && state_open && !panel_open)
+		if(loc == usr.loc)
+			to_chat(usr, span_warning("You can't close [src] on yourself!"))
```
Comment (Ghommie) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:406]:
```suggestion
		if(loc == user.loc)
			to_chat(user, span_warning("You can't close [src] on yourself!"))
```

**Review by Ghommie — APPROVED** [review]

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm (hunk 1 above)
Comment (Momo8289) [reply code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:406]: I had it as this initially, but was getting an error where `user` was always null. I'll try it again but if it still doesn't work, thoughts on how to fix that?

**Pull request merged**